metals/src/main/scala/scala/meta/internal/tvp/MetalsTreeViewProvider.scala

```diff
@@ -299,18 +317,16 @@ class MetalsTreeViewProvider(
     } yield TreeViewNode(
       Compile,
       id.getUri,
-      s"${info.getDisplayName()} - ${compilation.timer.toStringSeconds} (${compilation.progressPercentage}%)"
+      s"${info.getDisplayName()} - ${compilation.timer.toStringSeconds} (${compilation.progressPercentage}%)",
+      icon = "compile"
     )
   }
 
   private def ongoingCompilationNode: TreeViewNode = {
-    val size = compilations().size
-    val counter = if (size > 0) s" ($size)" else ""
     TreeViewNode(
       Compile,
-      "metals://ongoing-compilations",
```

Comment:
@ckipp01 Maybe this part is a problem with the `undefined` ? I hoped the node is discovered by id, but maybe we check something else in vim? This is the only big change, the rest is just moving things around

Reply:
No, I found the issue. The TVP panel in coc-metals was sort of built with the assumption that apart from the initial view it was never empty. Not that things have been shifted around the the compilations view will return an empty array just isn't accounted for. So I need to adjust the logic for this since I have logic built-in to find the offset of node to update, which is wrong now.

Reply:
Ok, cool. Thanks for taking a look at it!